File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int st = -1, en = nums.size(), i = 0;

        while (i != en) {
            if (nums[i] == 0 && i != st) swap(nums[i], nums[++ st]);
            else if (nums[i] == 2) swap(nums[i], nums[-- en]);
            else i += 1;
        }
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        st = dict()

        for i in range(len(nums)):
            if nums[i] in st and i - st.get(nums[i]) <= k: return True
            st[nums[i]] = i
        
        return False